main.c

```diff
@@ -0,0 +1,100 @@
+#include <signal.h>
+#include <stdio.h>
+
+#include "dns_sniffer.h"
+#include "firewall.h"
+
+#define IPV4 4
+#define IPV6 6
+
+#define NFLOG_GROUP 3
+
+#define SUCCESS_EXIT 0
+#define FAILURE_EXIT 1
+
+dns_sniffer_t sniffer;
```

Comment:
if really needed to be hold by dns_sniffer client, should be static and should be renamed g_sniffer or such.
note for dns_sniffer functions with the variable names sniffer